### ISSUE TITLE
zig build with invalid argument panics on macOS

### ISSUE THREAD

**Issue by euantorano** [ziglang/zig#1879]
Related to #1878 

Running `zig build -h` (`-h` is an invalid argument - it should be `--help`) results in a runtime panic on macOS Mojave:

```
Unrecognized argument: -h

Usage: /Users/euantorano/Downloads/zig/zig build [steps] [options]

Steps:
  default Build the project
  test Run library tests
  install Copy build artifacts to prefix path
  uninstall Remove build artifacts from prefix path

General Options:
  --help                 Print this help and exit
  --verbose              Print commands before executing them
  --prefix [path]        Override default install prefix
  --search-prefix [path] Add a path to look for binaries, libraries, headers

Project-Specific Options:
  -Drelease-safe=[bool] optimizations on and safety on
  -Drelease-fast=[bool] optimizations on and safety off
  -Drelease-small=[bool] size optimizations on and safety off

Advanced Options:
  --build-file [file]    Override path to build.zig
  --cache-dir [path]     Override path to zig cache directory
  --verbose-tokenize     Enable compiler debug output for tokenization
  --verbose-ast          Enable compiler debug output for parsing into an AST
  --verbose-link         Enable compiler debug output for linking
  --verbose-ir           Enable compiler debug output for Zig IR
  --verbose-llvm-ir      Enable compiler debug output for LLVM IR
  --verbose-cimport      Enable compiler debug output for C imports
error: InvalidArgs
integer overflow
/Users/euantorano/Downloads/zig/lib/zig/std/mem.zig:802:16: 0x10708672d in _join.46 (build.o)
        if (buf[buf_index - 1] != sep) {
               ^
/Users/euantorano/Downloads/zig/lib/zig/std/os/path.zig:51:5: 0x1070863a5 in _joinPosix (build.o)
    return mem.join(allocator, sep_posix, paths);
    ^
/Users/euantorano/Downloads/zig/lib/zig/std/os/path.zig:42:9: 0x1070862f9 in _join (build.o)
        return joinPosix(allocator, paths);
        ^
/Users/euantorano/Downloads/zig/lib/zig/std/debug/index.zig:1333:31: 0x107081dc3 in _LineNumberProgram_checkLineMatch (build.o)
            const file_name = try os.path.join(self.file_entries.allocator, dir_name, file_entry.file_name);
                              ^
/Users/euantorano/Downloads/zig/lib/zig/std/debug/index.zig:1678:17: 0x10707f689 in _getLineNumberInfoMacOs (build.o)
            if (try prog.checkLineMatch()) |info| return info;
                ^
/Users/euantorano/Downloads/zig/lib/zig/std/debug/index.zig:594:5: 0x10707cafb in _printSourceAtAddressMacOs (build.o)
    if (getLineNumberInfoMacOs(di, symbol.*, adjusted_addr)) |line_info| {
    ^
/Users/euantorano/Downloads/zig/lib/zig/std/debug/index.zig:266:30: 0x10707c828 in _printSourceAtAddress (build.o)
        builtin.Os.macosx => return printSourceAtAddressMacOs(debug_info, out_stream, address, tty_color),
                             ^
/Users/euantorano/Downloads/zig/lib/zig/std/debug/index.zig:197:33: 0x107070422 in _writeStackTrace (build.o)
        try printSourceAtAddress(debug_info, out_stream, return_address, tty_color);
                                ^
/Users/euantorano/Downloads/zig/lib/zig/std/debug/index.zig:102:20: 0x10706f5f6 in _dumpStackTrace (build.o)
    writeStackTrace(stack_trace, stderr, getDebugInfoAllocator(), debug_info, wantTtyColor()) catch |err| {
                   ^

Build failed. The following command failed:
/Users/euantorano/Library/Application Support/zig/stage1/artifact/I_nNjbDTlZR8HGRByCnjQbPwWkQxKXpo1lSPsznlbIGkaIlegHkwsUKvfljXMjHs/build /Users/euantorano/Downloads/zig/zig /Users/euantorano/Desktop/zig-crash-test /Users/euantorano/Desktop/zig-crash-test/zig-cache -h
```

**Comment by kristate** [comment]
Hmm, I am not seeing this on the @connectFree fork of zig.
I think we already solved this with #1786 and #1807

**Comment by euantorano** [comment]
Possibly. I’m using the most recent master build at the moment, so I’d be missing those changes.

**Comment by emekoi** [comment]
could we change build_runner so errors don't leak out except through error messages? when supplying an invalid option to build you don't need a stack trace of the build executable.

**Comment by kristate** [comment]
@emekoi Yes, this would be possible with `catch`.
I personally like that the invalid option triggers a stack trace so that the internals can be seen and understood -- the more that everyone can understand zig internals, the better chance we have to grow the language.

**Comment by fibric** [comment]
To me, its inconsistent behavior between CLI sees an `unrecognized command` and CLI sees an `unrecognized argument` for commands.

No stack trace when a command isn't recognized:
```shell
$ zig fmt                                                                                                                                                             
Unrecognized command: fmt
See `zig help` for detailed usage information
```
No stack trace when an argument isn't recognized:
```shell
$ zig -h                                                                                                                                                                
Expected another argument after -h
See `zig help` for detailed usage information
```
Stack trace printed when an argument for a command isn't recognized:
```shell
$ zig build -h                                                                                                                                                            
Unrecognized argument: -h
<...prints usage...>
<...prints stack trace...>
```

**Comment by andrewrk** [comment]
Fixed by @shritesh in #1938
